crates/iota-core/src/congestion_tracker.rs

```diff
@@ -169,19 +215,60 @@ impl CongestionTracker {
                 .map(|obj| obj.id),
         )
     }
+
+    /// For all the mutable shared inputs, sum the hotness of the objects.
+    /// More sophisticated prediction algorithms can be implemented.
+    pub fn get_suggested_gas_price_with_ogd(&self, transaction: &TransactionData) -> Option<u64> {
+        let hotness = self
+            .get_total_hotness_for_objects(
+                transaction
+                    .shared_input_objects()
+                    .into_iter()
+                    .filter(|id| id.mutable)
+                    .map(|id| id.id),
+            )
+            .unwrap_or(0.0);
+
+        Some(self.reference_gas_price + hotness as u64)
+    }
```

Comment:
The function does not need to return an `Option`, and can be rewritten as follows:

```suggestion
    pub fn get_suggested_gas_price_with_ogd(&self, transaction: &TransactionData) -> u64 {
        let hotness = self
            .get_total_hotness_for_objects(
                transaction
                    .shared_input_objects()
                    .into_iter()
                    .filter(|obj| obj.mutable)
                    .map(|obj| obj.id),
            );

        self.reference_gas_price + hotness as u64
    }
```

BTW, the function is never used? Where is it planned to be called?

Reply:
This function is supposed to be replacing Sui's estimation (`get_suggested_gas_price`) in `dry_exec_transaction_impl`

Reply:
Okay, then it should be invoked there? Because `dry_exec_transaction_impl` is still using Sui's way.